client/src/js/components/bhCurrencySelect.js

```diff
@@ -0,0 +1,98 @@
+angular.module('bhima.components')
+.component('bhCurrencySelect', {
+  controller : bhCurrencySelect,
+  templateUrl : 'partials/templates/bhCurrencySelect.tmpl.html',
+  bindings : {
+    validationTrigger : '<',
+    currencyId : '=',
+    disableIds : '<',
+    onChange : '&'
+  }
+});
+
+bhCurrencySelect.$inject = [ '$scope', 'CurrencyService' ];
```

Comment:
@jniles , I have an idea and I think it good.
When reading the component documentation, I understood that, the component is only responsible of his own view, that is why is scope is always isolated.
In theory you can access the entire view of the page where a component is injected by using  ```scope``` and ```watch``` but this is a bad practice.
When improving **currency input**,  I faced the same problem, I was supposed to use ```scope``` but I have an other alternative, **using the component tree design**.
So for the **currency input** for exemple, I will need the currency select first, so when a user change the currency, the currency select component will call an appropriate method of currency input. so the currency input will update his view and apply specific validation rule.
I will use your code (currency select) to perform it. And will submit the PR this evening, so you will review it.